`#undef WITH_THREAD` not supported

When `WITH_THREAD` is `#undef`ed in `$PYTHON_PREFIX/include/python2.7/pyconfig.h`.

    ./pybind11/include/pybind11/pybind11.h: In function ‘void pybind11::init_threading()’:
    ./pybind11/include/pybind11/pybind11.h:915:51: error: ‘PyEval_InitThreads’ was not declared in this scope
     inline void init_threading() { PyEval_InitThreads(); }


This problem is also probably in python3.x

How I got here? 3rd party custom python and I cannot change it.

--- a/include/pybind11/pybind11.h
+++ b/include/pybind11/pybind11.h
@@ -983,27 +983,14 @@ template <typename InputType, typename OutputType> void implicitly_convertible()
     ((detail::type_info *) it->second)->implicit_conversions.push_back(implicit_caster);
 }
 
-inline void init_threading() {
 #if defined(WITH_THREAD)
-  PyEval_InitThreads();
-#endif
-}
+inline void init_threading() { PyEval_InitThreads(); }
 
 class gil_scoped_acquire {
-#if defined(WITH_THREAD)
     PyGILState_STATE state;
-#endif
 public:
-    inline gil_scoped_acquire() {
-#if defined(WITH_THREAD)
-      state = PyGILState_Ensure();
-#endif
-    } 
-    inline ~gil_scoped_acquire() {
-#if defined(WITH_THREAD)
-      PyGILState_Release(state); 
-#endif
-    }
+    inline gil_scoped_acquire() { state = PyGILState_Ensure(); } 
+    inline ~gil_scoped_acquire() { PyGILState_Release(state); }
 };
 
 class gil_scoped_release {
@@ -1012,6 +999,7 @@ public:
     inline gil_scoped_release() { state = PyEval_SaveThread(); }
     inline ~gil_scoped_release() { PyEval_RestoreThread(state); }
 };
+#endif
 
 inline function get_overload(const void *this_ptr, const char *name)  {
     handle py_object = detail::get_object_handle(this_ptr);

--- a/include/pybind11/pybind11.h
+++ b/include/pybind11/pybind11.h
@@ -1004,9 +1004,6 @@ public:
       PyGILState_Release(state); 
 #endif
     }
-#if defined(WITH_THREAD)
-  PyEval_InitThreads();
-#endif
 };
 
 class gil_scoped_release {

--- a/include/pybind11/pybind11.h
+++ b/include/pybind11/pybind11.h
@@ -983,13 +983,30 @@ template <typename InputType, typename OutputType> void implicitly_convertible()
     ((detail::type_info *) it->second)->implicit_conversions.push_back(implicit_caster);
 }
 
-inline void init_threading() { PyEval_InitThreads(); }
+inline void init_threading() {
+#if defined(WITH_THREAD)
+  PyEval_InitThreads();
+#endif
+}
 
 class gil_scoped_acquire {
+#if defined(WITH_THREAD)
     PyGILState_STATE state;
+#endif
 public:
-    inline gil_scoped_acquire() { state = PyGILState_Ensure(); }
-    inline ~gil_scoped_acquire() { PyGILState_Release(state); }
+    inline gil_scoped_acquire() {
+#if defined(WITH_THREAD)
+      state = PyGILState_Ensure();
+#endif
+    } 
+    inline ~gil_scoped_acquire() {
+#if defined(WITH_THREAD)
+      PyGILState_Release(state); 
+#endif
+    }
+#if defined(WITH_THREAD)
+  PyEval_InitThreads();
+#endif
 };
 
 class gil_scoped_release {

